getconf/base.py

```diff
@@ -192,6 +193,31 @@ def getpath(self, key, default=Path('.'), doc=''):
             logger.exception("Unable to cast %r as Path for the key %s.", value, key)
             raise
 
+    def getenum(self, key, default=None, doc='', enum_class=None):
+        if enum_class is None and isinstance(default, enum.Enum):
+            enum_class = type(default)
+        assert (
+            default is not None and enum_class is not None and issubclass(enum_class, enum.Enum) and issubclass(enum_class, type(default))
+        ), 'getenum("%s", %r, enum_class=%r) is missing a default value or this one has an invalid type.' % (
```

Comment:
I don't think the default value should be mandatory if an `enum_class` is provided.

Reply:
Maybe we should stick with `None` in such case then ?
```
In [9]: CONFIG.getstr('foo')
Out[9]: ''

In [10]: CONFIG.getstr('foo', default=None)

In [11]: CONFIG.getint('foo')
Out[11]: 0

In [12]: CONFIG.getint('foo', default=None)
```

To follow the same behavior, I'd expect `CONFIG.getenum('foo', default=None, enum_class=TestIntEnum)` to return `None` if no configured value is found.

I agree that we would be missing a consistent behavior for `CONFIG.getenum('foo', enum_class=TestIntEnum)` but also returning `None` in such case seems like an acceptable default.